### PR TITLE
feat(async-csv): Limit the number of columns for exports to 20

### PR DESCRIPTION
**Pull request by Zylphrex**

### CODE REVIEW

**Review by wmak — APPROVED** [review]
nice

File: src/sentry/data_export/endpoints/data_export.py [hunk 1]
```diff
@@ -47,6 +48,12 @@ def post(self, request, organization):
             return Response(serializer.errors, status=400)
         data = serializer.validated_data
 
+        if len(data["query_info"].get("field", [])) > MAX_FIELDS:
+            detail = "You can export up to {0} fields at a time. Please delete some and try again.".format(
+                MAX_FIELDS
+            )
+            return Response({"detail": detail}, status=400)
```
Comment (markstory) [inline src/sentry/data_export/endpoints/data_export.py:55]: You could use `rest_framework.exceptions.ParseError` for these.

Comment (Zylphrex) [reply src/sentry/data_export/endpoints/data_export.py]: Thanks for the tip! I updated it.

**Review by markstory — APPROVED** [review]

**Comment by Zylphrex** [comment]
I remembered that issues by tag exports also use this same end point so it was causing issues with tests. Moving the check to discover only exports.

**Pull request merged**